tools/bin/mbedtls-run-tests

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+"""Run Mbed TLS unit tests.
+
+By default, run all test casess. If at least one of --prefix, --regex or
+--substring is specified, only run the test cases that match any of these
+options.
+"""
+
+import argparse
+import os
+import re
+import subprocess
+import tempfile
+
+
+class TestCaseFilter:
+    """Test case filter."""
+
+    def __init__(self, options):
+        """Set up a test case filter.
+
+        See `main` for what options are valid.
+        """
+        include = ''
+        for string in options.substring:
+            include += '|.*' + re.escape(string)
+        for string in options.prefix:
+            include += '|' + re.escape(string)
+        for regex in options.regex:
+            include += '|.*' + regex
+        if include: # no filter specified
+            include = include[1:] # remove the leading '|'
+        self.include = re.compile(include.encode())
+        if options.exclude:
+            exclude = '|'.join(options.exclude)
+        else:
+            exclude = r'.\A' # matches nothing
+        self.exclude = re.compile(exclude.encode())
+
+    def match(self, description):
+        """Whether the given test case description matches the filter."""
+        return (re.match(self.include, description) and
+                not re.match(self.exclude, description))
+
+
+def extract_description(stanza):
+    """Extract the description from a .data stanza."""
+    m = re.match(r'(?:\n|#[^\n]*\n)*([^#\n][^\n]*)\n', stanza + '\n')
+    if m:
+        return m.group(1)
+    else:
+        return None
+
+def filter_test_cases(all_datax, tcf, temp_datax):
+    """Filter test cases.
+
+    Filter test cases from datax_file based on their description according
+    to the filter tcf. Write the selected test cases to temp_datax.
+    """
+    in_stanza = False
+    in_match = False
+    for line in open(all_datax, 'rb'):
```

Comment:
Possibly nicer to put all this in a "with open ... " block and then the temp_datax.flush() at the end is redundant

Reply:
`temp_datax` is already coming from the `with tempfile.NamedTemporaryFile` in the caller. How would using a `with open` for `all_datax` help?